tensorflow_probability/python/sts/components/autoregressive_moving_average.py

```diff
@@ -0,0 +1,238 @@
+        """
+        parameters = dict(locals())
+        parameters.update(linear_gaussian_ssm_kwargs)
+        del parameters['linear_gaussian_ssm_kwargs']
+        with tf.name_scope(name or 'ARMAStateSpaceModel') as name:
+            # The initial state prior determines the dtype of sampled values.
+            # Other model parameters must have the same dtype.
+            dtype = initial_state_prior.dtype
+
+            ar_coefficients = tf.convert_to_tensor(
+                value=ar_coefficients, name='ar_coefficients', dtype=dtype)
+            ma_coefficients = tf.convert_to_tensor(
+                value=ma_coefficients, name='ma_coefficients', dtype=dtype)
+            level_scale = tf.convert_to_tensor(
+                value=level_scale, name='level_scale', dtype=dtype)
+            observation_noise_scale = tf.convert_to_tensor(
+                value=observation_noise_scale,
+                name='observation_noise_scale', dtype=dtype)
+
+            p = tf.compat.dimension_value(ar_coefficients.shape[-1])
+            q = tf.compat.dimension_value(ma_coefficients.shape[-1])
+
+            if p is None or q is None:
+                raise ValueError('coefficients must have static shape.')
+
+            # if p != q + 1, then pad either of the arguments to ensure
+            # we end up fitting an ARMA(p, p-1) process.
+            if p > q + 1:
+                pad = tf.zeros([p - q - 1], dtype=dtype)
+                ma_coefficients = tf.concat([ma_coefficients, pad], axis=-1)
+            elif q + 1 > p:
+                pad = tf.zeros([q + 1 - p], dtype=dtype)
+                ar_coefficients = tf.concat([ar_coefficients, pad], axis=-1)
+
+            order = max(p, q + 1)
+            if order is None:
+                raise ValueError('coefficients must have static shape.')
+
+            self._order = order
+            self._ar_coefficients = ar_coefficients
+            self._ma_coefficients = ma_coefficients
+            self._level_scale = level_scale
+
+            super(AutoregressiveMovingAverageStateSpaceModel, self).__init__(
+                num_timesteps=num_timesteps,
+                transition_matrix=make_ar_transition_matrix(ar_coefficients),
+                transition_noise=tfd.MultivariateNormalDiag(
+                    scale_diag=tf.stack([level_scale] +
+                                        [tf.zeros_like(level_scale)] * (
+                                                self.order - 1), axis=-1)),
+                observation_matrix=make_ma_observation_matrix(ma_coefficients),
+                observation_noise=tfd.MultivariateNormalDiag(
+                    scale_diag=observation_noise_scale[..., tf.newaxis]),
+                initial_state_prior=initial_state_prior,
+                name=name,
+                **linear_gaussian_ssm_kwargs)
+            self._parameters = parameters
+
+    @property
+    def order(self):
+        return self._order
+
+    @order.setter
```

Comment:
Is this setter needed/used anywhere? We generally don't allow mutation of public attributes on TFP objects.